src/Function.h

```diff
@@ -68,6 +68,10 @@ class Function {
     /** Construct a Function from an existing FunctionContents pointer. Must be non-null */
     explicit Function(const FunctionPtr &);
 
+    /** Construct a function from deserializing */
```

Comment:
The arguments to this need documentation:
- How is origin_name different from name?
- How are required_types different from output_types?
- What kind of strings are ok for args?
- What happens if I pass empty vectors for any of the arguments?